mlflow/langchain/__init__.py

```diff
@@ -726,6 +767,24 @@ def load_model(model_uri, dst_path=None):
     return _load_model_from_local_fs(local_model_path)
 
 
+def _load_code_model(local_model_path, flavor_conf, isLocal=False):
+    databricks_rag_config = flavor_conf.get(_CODE_CONFIG)
+    flavor_code_config = flavor_conf.get(FLAVOR_CONFIG_CODE) or ""
+    code_paths = {
+        path: os.path.join(
+            local_model_path,
+            flavor_code_config,
+            path if isLocal else os.path.basename(os.path.abspath(path)),
+        )
+        for path in databricks_rag_config
+    }
+    _set_config_path(code_paths)
+
+    import chain  # noqa: F401
```

Comment:
I would expect to add the code path to the system path before performing this import. I think that should be done within this method. See also https://github.com/mlflow/mlflow/pull/11370/files#r1520525504.

Reply:
In loading the model back, we are adding the entire local_path to sys path: https://github.com/mlflow/mlflow/pull/11370/files/06759dae66a46a4fb753f6e72612a37753585c83#diff-44b09327d1292c38e7ce251f66635a35c22720697ae1f39db6d4873d98109548R736